plugins/policy/configurator/configurator_impl.go

```diff
@@ -130,45 +137,61 @@ func (pct *PolicyConfiguratorTxn) Commit() error {
 	for pod, unorderedPolicies := range pct.config {
 		var ingress ContivRules
 		var egress ContivRules
+		var delPodConfig bool
 
 		// Get target pod configuration.
+		podIPNet, hadIPAddr := pct.podIPAddresses[pod]
 		found, podData := pct.configurator.Cache.LookupPod(pod)
-		if !found {
-			pct.Log.WithField("pod", pod).Warn("Pod data not found in the cache")
-			continue
-		}
 
-		// Get pod IP address (expressed as one-host subnet).
-		if podData.IpAddress == "" {
-			pct.Log.WithField("pod", pod).Warn("Pod has no IP address assigned")
-			continue
-		}
-		podIPNet := utils.GetOneHostSubnet(podData.IpAddress)
-		if podIPNet == nil {
-			pct.Log.WithField("pod", pod).Warn("Pod has invalid IP address assigned")
-			continue
+		// Handle removed pod.
+		if !found || podData.IpAddress == "" {
+			if hadIPAddr {
+				pct.Log.WithField("pod", pod).Debug("Removing policies from the pod.")
+				delPodConfig = true
+				delete(pct.podIPAddresses, pod)
+			} else {
+				/* already un-configured */
+				continue
+			}
 		}
 
-		// Sort policies to get the same outcome for the same set.
-		policies := unorderedPolicies.Copy()
-		sort.Sort(policies)
-
-		// Check if this set was already processed.
-		alreadyProcessed := false
-		for _, policySet := range processed {
-			if policySet.policies.Equals(policies) {
-				ingress = policySet.ingress
-				egress = policySet.egress
-				alreadyProcessed = true
+		if !delPodConfig {
+			// Get pod IP address (expressed as one-host subnet).
+			podIPNet = utils.GetOneHostSubnet(podData.IpAddress)
+			if podIPNet == nil {
+				pct.Log.WithField("pod", pod).Warn("Pod has invalid IP address assigned")
+				continue
+			}
+			pct.podIPAddresses[pod] = podIPNet
+
+			// Sort policies to get the same outcome for the same set.
+			policies := unorderedPolicies.Copy()
+			sort.Sort(policies)
+
+			// Check if this set was already processed.
+			alreadyProcessed := false
+			for _, policySet := range processed {
```

Comment:
This is trying to remove duplicated policies, i think we could do a pre-processing of `unorderedPolicies` to make it in order and without duplication, but it's fine for now